### ISSUE TITLE
SIP Finetune on CheXpert - Validation sanity check error - target has to be an integer tensor

### ISSUE THREAD

**Issue by silgonel** [facebookresearch/CovidPrognosis#15]
Hello,
I am trying to finetune with train_sip.py and the public dataset CheXpert.
At the moment, I simply launched the training by specifying one of the pretrained model available online and the dataset name and directory. Is there any other configuration step that should be made?

At the moment, i am getting the error "target has to be an integer tensor" during _basic_input_validation (see log below).
When going back to the function validation_epoch_end ("sip_finetune.py", line 213), i noticed that "targets" is defined as a Tensor - dtype = float32 , while "logits" values are in the range [-1,1]. I was expecting them to be between 0 and 1... is that right?
What could I do to fix this?

Thank you in advance!
Silvia

**LOG:**
_Validation sanity check:  50%|█████     | 1/2 [00:00<00:00,  1.40it/s]path: Atelectasis, len: 64
Traceback (most recent call last):

  File "/Facebook_NYU/CovidPrognosis-master/cp_examples/sip_finetune/train_sip.py", line 205, in <module>
    cli_main(args)

[...]
  File "/Facebook_NYU/CovidPrognosis-master/cp_examples/sip_finetune/sip_finetune.py", line 231, in validation_epoch_end
    self.val_acc[i](logits, targets)

  File "/home/anaconda3/envs/facebook/lib/python3.9/site-packages/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)

  File "/home/anaconda3/envs/facebook/lib/python3.9/site-packages/torchmetrics/metric.py", line 152, in forward
    self.update(*args, **kwargs)

  File "/home/anaconda3/envs/facebook/lib/python3.9/site-packages/torchmetrics/metric.py", line 199, in wrapped_func
    return update(*args, **kwargs)

  File "/home/anaconda3/envs/facebook/lib/python3.9/site-packages/pytorch_lightning/metrics/classification/accuracy.py", line 138, in update
    correct, total = _accuracy_update(

  File "/home/anaconda3/envs/facebook/lib/python3.9/site-packages/pytorch_lightning/metrics/functional/accuracy.py", line 25, in _accuracy_update
    preds, target, mode = _input_format_classification(preds, target, threshold=threshold, top_k=top_k)

  File "/home/anaconda3/envs/facebook/lib/python3.9/site-packages/pytorch_lightning/metrics/classification/helpers.py", line 433, in _input_format_classification
    case = _check_classification_inputs(

  File "/home/anaconda3/envs/facebook/lib/python3.9/site-packages/pytorch_lightning/metrics/classification/helpers.py", line 296, in _check_classification_inputs
    _basic_input_validation(preds, target, threshold, is_multiclass)

  File "/home/anaconda3/envs/facebook/lib/python3.9/site-packages/pytorch_lightning/metrics/classification/helpers.py", line 62, in _basic_input_validation
    raise ValueError("The `target` has to be an integer tensor.")

**ValueError: The `target` has to be an integer tensor.**_

**Comment by mmuckley** [comment]
Hello @silgonel.
First, I apologize for taking so long to respond. I was out of work for a bit and have been working on some other things.

The logits are the log probabilities, so their range is the entire real number line. You run a sigmoid layer to get the probabilities.

As for the error, I can't reproduce it. On my machine I ran the following command:

`python train_sip.py --pretrained_file https://dl.fbaipublicfiles.com/CovidPrognosis/pretrained_models/mimic_lr_0.1_bs_128_fd_128_qs_65536.pt --dataset_name chexpert`

I put the location of my CheXpert data in [this file](https://github.com/facebookresearch/CovidPrognosis/blob/master/configs/data.yaml).

I can complete a full training epoch with training and validation.

Do these steps mirror those you took?

**Comment by silgonel** [comment]
Hello @mmuckley ,

Thank you very much for your kind answer!
My colleague  @nataschadam and I followed your steps, but we are still getting the same error. 
We are currently using the "Chexpert v1.0-small" version of the dataset (the one with reduced size of 11GB).
Maybe you are testing the code with the full-size dataset? Could there be any difference?

Many thanks,
Silvia

**Comment by mmuckley** [comment]
@silgonel 
What version of PyTorch and PyTorch Lightning are you running? It looks like this error was thrown by Lightning. I was on 1.7.1 and 1.0.8.

**Comment by mmuckley** [comment]
@silgonel
Looking at your requirements.txt it looks like you're using PyTorch Lightning 1.2.7 while the repository was tested on 1.0.8. Unfortunately, I'm tied up with other projects at the moment so I don't think I can update the repository to the new version. If you'd like to reproduce the results, I would roll back PyTorch Lightning to 1.0.8. Alternatively, you could fork the repository and update the relevant lines to output integer tensors and use the new version.

**Comment by nataschadam** [comment]
Hello @mmuckley , 
I am a colleague of @silgonel . 

Thank you very much for your help. After the answer you gave us yesterday, we realized that even if we used the file "requirements" to create a new conda environment, conda didn't install the required versions of the packages. We created a new conda environment manually installing the required packages, and fortunately, the problem is gone and the training is working. 
Thank you very much for your help!
